### PR TITLE
Use compile_assert function instead of assert keyword

### PR DESCRIPTION
**Pull request by jdahm**
## Description

Using the `assert` keyword for compile-time assertions causes issues when running with the `PYTHONOPTIMIZE=True` env var or `-O` command line arguments. This PR switches gtscript to a builtin `external_assert` gtscript function, to get around the issue.

To do:
- [x] Unit tests

### CODE REVIEW

File: src/gt4py/frontend/gtscript_frontend.py [hunk 1]
```diff
@@ -579,9 +589,9 @@ def visit_Call(self, node: ast.Call, *, target_node=None):
         return result_node
 
     def visit_Expr(self, node: ast.Expr):
-        """ignore pure string statements in callee"""
-        if not isinstance(node.value, ast.Str):
-            return super().visit(node)
+        """ignore pure string statements in callee."""
```
Comment (egparedes) [inline src/gt4py/frontend/gtscript_frontend.py:592]:
```suggestion
        """Ignore pure string statements in callee."""
```

**Review by egparedes — COMMENTED** [review]
Looks good, but I find the name `external_assert` a bit misleading. What's the reason for this name? Wouldn't you prefer something more explicit like `static_assert` or `compile_assert`?

**Comment by jdahm** [comment]
> What's the reason for this name?

Everyone seems to have this question ;) It came up because the assert can only be on constants which are `externals`.

> Wouldn't you prefer something more explicit like static_assert or compile_assert?

`static_assert` could conflict with mypy type checking https://github.com/python/mypy/issues/5687. I thought about `compile_assert`. We could change to that, but `external_assert` is already scattered throughout fv3core...

**Comment by egparedes** [comment]
Thanks for the explanation. I still think that `external_assert` is a bad name and it should be changed, but I don't want to force any changes in `fv3core`, so I'm not sure what's the best decision here. What do you think about waiting until you have time to change the code in `fv3core` and merge later?

**Comment by jdahm** [comment]
We could switch to `compile_assert` when we switch to `Field[IJK, data_type]`.

**Comment by jdahm** [comment]
This is ready to go now.

**Review by egparedes — APPROVED** [review]
It looks good. Thanks for contributing.

**Pull request merged**